aruco_ros/src/aruco_detector.cpp

```diff
@@ -189,14 +228,65 @@ class ArucoSimple
         //Ok, let's detect
         mDetector.detect(inImage, markers, camParam, marker_size, false);
         //for each marker, draw info and its boundaries in the image
+
+        bool multiple_markers_detected = markers.size() > 1;
+        unsigned int error_condition = 0;
+        std::string error_message = "";
+
         for(size_t i=0; i<markers.size(); ++i)
         {
 
-          if (markers.size() == 1){  // Only process markers if there is only one in FOV
+          if (markers.size() == 1){
+            // Only process markers if there is only one in FOV
             // only publishing the selected markers
             if(is_marker_id_in_list(markers[i].id))
             {
+
               tf::Transform transform = aruco_ros::arucoMarker2Tf(markers[i]);
+              double roll, pitch, yaw;
+
+              tf::Matrix3x3(transform.getRotation()).getRPY(roll, pitch, yaw);
+
+              // Test for erroneous conditions
+              // Note that order is important for error message
+              // All error codes will be forwarded
+              if (abs(yaw - expected_yaw) > yaw_tolerance){
+                error_message = arucoMsg.CODE_TWISTED_MESSAGE;
+                error_condition |= arucoMsg.CODE_TWISTED;
+              }
+              if (abs(yaw - expected_yaw) > M_PI/2.){
```

Comment:
Also, I would put the bad angle errors at the end, since only the last message will appear and it seems more frustrating to, say, move the robot closer and then be told that the code needs to be repositioned, rather than the other way around